gateway/crates/gateway-binary/src/main.rs

```diff
@@ -181,6 +179,9 @@ fn otel_layer_reload<S>(
         let ReloadableOtelLayers {
             tracer,
             meter_provider,
+            // I swear I don't really want to know what the compiler is doing here.
+            // Set this to `logger: _`, and the OTEL reload will explode.
```

Comment:
`logger: _` drops the logger instantly, which is probably the problem. But `logger: _something` wouldn't drop immediately, so that should work without the annotation.